### PR TITLE
Add build guide using Docker

### PR DESCRIPTION
**Pull request by segfaultxavi**

### CODE REVIEW

File: docs/BUILD-docker.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+python3 catapult-src/scripts/build/runDockerBuild.py \
+    --compiler-configuration catapult-src/scripts/build/configurations/gcc-10.yaml
+    --build-configuration catapult-src/scripts/build/configurations/release-private.yaml
+    --operating-system ubuntu
+    --user 1000:1000
+    --destination-image-label gcc-10-main-9273d6c5
+```
+
+Note the used ``--compiler-configuration`` and ``--operating-system`` parameters that build tools for **Ubuntu + gcc10**.
+
```
Comment (gimer) [inline docs/BUILD-docker.md:42]: add a note, that `--user` should map to local user id and git, actually bit better alternative might be:
`--user "$(id -u):$(id -g)"`

File: docs/README.md [hunk 2]
```diff
@@ -64,22 +70,24 @@ When building sanitizers, `sanitizer_blacklist.txt` file is used.
...
 
- * for false positive in libc++ `shared_ptr`:
- * for false positive in boost's 1.74 executor (ref-counted)
- * for false positive in boost logger, in server logger is always initialized from a single thread
+* for false positive in libc++ `shared_ptr`:
+* for false positive in boost's 1.74 executor (ref-counted)
+* for false positive in boost logger, in server logger is always initialized from a single thread
 
-```
+```suppresions
```
Comment (gimer) [inline docs/README.md:77]: is there syntax highlight support for suppresions?

**Review by gimer — APPROVED** [review]

File: docs/README.md (hunk 2 above)
Comment (segfaultxavi) [reply docs/README.md:77]: Actually, no. This was a experiment that slipped through.

File: docs/BUILD-docker.md (hunk 1 above)
Comment (segfaultxavi) [reply docs/BUILD-docker.md:42]: Did I get it right?

File: docs/BUILD-docker.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+   ```bash
+   python3 ./scripts/build/baseImageDockerfileGenerator.py \
+       --compiler-configuration scripts/build/configurations/gcc-10.yaml \
+       --operating-system ubuntu \
+       --versions ./scripts/build/versions.properties \
+       --layer test
+   ```
+
+The final image name is `symbolplatform/symbol-server-build-base:ubuntu-gcc-10-skylake`
```
Comment (Jaguar0625) [inline docs/BUILD-docker.md:183]: image name is dependent on configuration

**Review by Jaguar0625 — APPROVED** [review]

**Comment by Jaguar0625** [comment]
Merged in 1b8ec56b4.

**Pull request closed without merging**